parser-typechecker/src/Unison/Codebase/Path.hs

```diff
@@ -100,14 +102,13 @@ prefix (Absolute (Path prefix)) (Path' p) = case p of
 -- Left is some parse error tbd
 parsePath' :: String -> Either String Path'
 parsePath' p = case parsePathImpl' p of
-  Left  e       -> Left e
-  Right (p, "") -> Right p
-  Right (p, rem) ->
-    case (first show . (Lexer.wordyId0 <> Lexer.symbolyId0) <> unit') rem of
-      Right (seg, "") -> Right (unsplit' (p, NameSegment . Text.pack $ seg))
-      Right (_, rem) ->
-        Left ("extra characters after " <> show p <> ": " <> show rem)
-      Left e -> Left e
+  Left  e        -> Left e
```

Comment:
This should be replaced with Megaparsec sometime. The implementation is hard to understand.